From 5a7d79f96463c060448fb24349f323c82ef41a4f Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 20 Oct 2021 09:27:26 +0300
Subject: [PATCH 02/11] s32-gen1: emu: Adapt base addresses

Text and data base addresses has to be lowered on emulator
because only 1GB of DDR is available.

Issue: ALB-7712
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 board/freescale/s32-gen1/Kconfig | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/board/freescale/s32-gen1/Kconfig b/board/freescale/s32-gen1/Kconfig
index 7795ab960d..61792d4472 100644
--- a/board/freescale/s32-gen1/Kconfig
+++ b/board/freescale/s32-gen1/Kconfig
@@ -43,6 +43,7 @@ config CUSTOM_SYS_DATA_BASE
 
 config SYS_DATA_BASE
 	hex "Data Base" if CUSTOM_SYS_DATA_BASE
+	default 0xbfa00000 if S32_ATF_BOOT_FLOW && TARGET_TYPE_S32GEN1_EMULATOR
 	default 0xffa00000 if S32_ATF_BOOT_FLOW
 	default 0x34000000
 	help
@@ -57,6 +58,7 @@ config CUSTOM_SYS_TEXT_BASE
 
 config SYS_TEXT_BASE
 	hex "Text Base" if CUSTOM_SYS_TEXT_BASE
+	default 0xbfaa0000 if S32_ATF_BOOT_FLOW && TARGET_TYPE_S32GEN1_EMULATOR
 	default 0xffaa0000 if S32_ATF_BOOT_FLOW
 	default 0x340a0000
 
@@ -106,6 +108,7 @@ config CUSTOM_DTB_SRAM_ADDR
 
 config DTB_SRAM_ADDR
 	hex "SRAM address at which the dtb will be found" if CUSTOM_DTB_SRAM_ADDR
+	default 0xbfa90000 if S32_ATF_BOOT_FLOW && TARGET_TYPE_S32GEN1_EMULATOR
 	default 0xffa90000 if S32_ATF_BOOT_FLOW
 	default 0x34090000
 
-- 
2.17.1

